osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetGsmDiagnosticCommandExecutorTest.java

```diff
@@ -0,0 +1,263 @@
+    when(result7.getResultCode()).thenReturn(AccessResultCode.SUCCESS);
+
+    final DataObject operator = mock(DataObject.class);
+    when(result2.getResultData()).thenReturn(operator);
+    when(operator.getValue()).thenReturn(new byte[] {65, 66});
+
+    final DataObject modemRegistrationStatus = mock(DataObject.class);
+    when(result3.getResultData()).thenReturn(modemRegistrationStatus);
+    when(modemRegistrationStatus.getValue()).thenReturn(2);
+
+    final DataObject csStatus = mock(DataObject.class);
+    when(result4.getResultData()).thenReturn(csStatus);
+    when(csStatus.getValue()).thenReturn(3);
+
+    final DataObject psStatus = mock(DataObject.class);
+    when(result5.getResultData()).thenReturn(psStatus);
+    when(psStatus.getValue()).thenReturn(4);
+
+    final DataObject cellInfo = mock(DataObject.class);
+    when(result6.getResultData()).thenReturn(cellInfo);
+    final DataObject cellId = mock(DataObject.class);
+    final DataObject locationId = mock(DataObject.class);
+    final DataObject signalQuality = mock(DataObject.class);
+    final DataObject bitErrorRate = mock(DataObject.class);
+    final DataObject mcc = mock(DataObject.class);
+    final DataObject mnc = mock(DataObject.class);
+    final DataObject channelNumber = mock(DataObject.class);
+    when(cellInfo.getValue())
+        .thenReturn(
+            Arrays.asList(
+                cellId, locationId, signalQuality, bitErrorRate, mcc, mnc, channelNumber));
+    when(cellId.getValue()).thenReturn(128L);
+    when(locationId.getValue()).thenReturn(1);
+    when(signalQuality.getValue()).thenReturn((short) 2);
+    when(bitErrorRate.getValue()).thenReturn((short) 3);
+    when(mcc.getValue()).thenReturn(4);
+    when(mnc.getValue()).thenReturn(5);
+    when(channelNumber.getValue()).thenReturn(6L);
+
+    final DataObject adjacentCells = mock(DataObject.class);
+    when(result7.getResultData()).thenReturn(adjacentCells);
+    final DataObject adjacentCell = mock(DataObject.class);
+    when(adjacentCells.getValue()).thenReturn(Collections.singletonList(adjacentCell));
+    final DataObject adjacentCellId = mock(DataObject.class);
+    final DataObject adjacentCellSignalQuality = mock(DataObject.class);
+    when(adjacentCell.getValue())
+        .thenReturn(Arrays.asList(adjacentCellId, adjacentCellSignalQuality));
+    when(adjacentCellId.getValue()).thenReturn(256L);
+    when(adjacentCellSignalQuality.getValue()).thenReturn((short) 7);
+
+    when(this.dlmsHelper.getAndCheck(
+            eq(this.connectionManager), eq(this.device), eq("Get GsmDiagnostic"), any()))
+        .thenReturn(Arrays.asList(result2, result3, result4, result5, result6, result7));
+
+    // CALL
+    final GetGsmDiagnosticResponseDto result =
+        this.executor.execute(this.connectionManager, this.device, this.request);
+
+    // VERIFY calls to mocks
+    final String classIdAndObis = "99,1.2.3.4.5.6";
```

Comment:
`classIdAndObis` should be used or removed.

Reply:
Removed